File: 263.ugly-number.go
/*
 * @lc app=leetcode id=263 lang=golang
 *
 * [263] Ugly Number
 */

// @lc code=start
func isUgly(n int) bool {
	for i := 2; i < 6 && n > 0; i++ {
		for n%i == 0 {
			n /= i
		}
	}

	return n == 1
}

// @lc code=end

File: 263.ugly-number.cpp
/*
 * @lc app=leetcode id=263 lang=cpp
 *
 * [263] Ugly Number
 */

// @lc code=start
class Solution {
 public:
  bool isUgly(int n) {
    int i = 2;
    while (n > 1) {
      while (n % i == 0) n /= i;

      ++i;

      if (i > 5) break;
    }

    return n == 1;
  }
};
// @lc code=end
